File: app/src/main/java/com/physphile/forbot/BaseSwipeActivity.java
package com.physphile.forbot;

import android.os.Bundle;
import android.widget.CompoundButton;

import androidx.annotation.Nullable;

import com.example.swipebacklib.BaseSwipeBackActivity;
import com.example.swipebacklib.SwipeBackLayout;

public abstract class BaseSwipeActivity extends BaseSwipeBackActivity implements CompoundButton.OnCheckedChangeListener {
    public boolean isMain = false;
    protected SwipeBackLayout mSwipeBackLayout;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
        mSwipeBackLayout = getSwipeBackLayout();
        if (isMain) {
            return;
        }
    }

    protected abstract int getLayoutId();
}

File: app/src/main/java/com/physphile/forbot/BaseSwipeActivity.kt
package com.physphile.forbot

import android.os.Bundle
import android.widget.CompoundButton
import com.example.swipebacklib.BaseSwipeBackActivity
import com.example.swipebacklib.SwipeBackLayout

abstract class BaseSwipeActivity : BaseSwipeBackActivity(), CompoundButton.OnCheckedChangeListener {
    var isMain = false
    protected var mSwipeBackLayout: SwipeBackLayout? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutId)
        mSwipeBackLayout = swipeBackLayout
        if (isMain) {
            return
        }
    }

    protected abstract val layoutId: Int
}